odoo/tools/cache.py

```diff
@@ -191,3 +221,32 @@ def get_cache_key_counter(bound_method: Callable, *args, **kwargs):
     cache, key0, counter = ormcache_instance.lru(model)
     key = key0 + ormcache_instance.key(model, *args, **kwargs)
     return cache, key, counter
+
+
+def get_memory_footprint(*args):
+    """ A non-thread-safe recursive object size estimator """
+    seen_ids = set()
+    total_size = 0
+    objects = list(args)
+
+    while objects:
+        cur_obj = objects.pop()
+        if id(cur_obj) in seen_ids or callable(cur_obj):
+            continue
+
+        seen_ids.add(id(cur_obj))
+        total_size += sys.getsizeof(cur_obj)
+
+        if hasattr(cur_obj, '__slot__'):
+            objects.extend(getattr(cur_obj, s) for s in cur_obj.__slots__ if hasattr(cur_obj, s))
```

Comment:
Does it work ? :astonished: 

`hasattr(cur_obj, '__slot__')`, do you mean `__slots__` ? Also, you shouldn't trust `__slots__` since `self.env['res.partner'].__slots__` returns a empty `()` but we actually have __slots__ on BaseModel.

Maybe you need to use `dir()` here and filter out methods.